internal/core/src/mmap/Utils.h

```diff
@@ -110,6 +111,18 @@ WriteFieldData(File& file,
                 }
                 break;
             }
+            case DataType::ARRAY: {
+                for (ssize_t i = 0; i < data->get_num_rows(); ++i) {
```

Comment:
for num_rows, array length, these are not negative, using size_t is more suiotable,  same rule below